### ISSUE TITLE
Pull request description for upgrade dependencies workflow

### ISSUE THREAD

**Issue by sgrell** [projen/projen#3002]
For our typescript application we configured the dependency upgrade workflow. We configured the title of the pull request and the schedule of the workflow, but I can't find any option to configure the description of the pull request, it is always the fixed text:

> Upgrades project dependencies. See details in [workflow run].
>
> [Workflow Run]: https://github.com/...
>
> ------
>
> *Automatically created by projen via the "upgrade" workflow*

Since we have a few manuell steps that are required before the merge, it is very important for us to adapt the pr description and add these steps to make them harder to forget.

We also have a pr template configured that contains all the necessary information, but it is not used for dependency updates.

Our current configuration looks like this:

```typescript
const project = new awscdk.AwsCdkTypeScriptApp({
  ...
  depsUpgrade: true,
  depsUpgradeOptions: {
    pullRequestTitle: 'Dependency update',
    workflowOptions: {
      projenCredentials: github.GithubCredentials.fromApp(),
      schedule: UpgradeDependenciesSchedule.WEEKLY,
    },
  },
  ...
});

```

Is there already a way to configure this? If not, would it be possible to add an option for it?

**Comment by mrgrain** [comment]
Good feature request, should be supported.

In the meantime you can use[ low-level object file manipulation features](https://projen.io/escape-hatches.html) to change the text.

**Comment by sgrell** [comment]
@mrgrain thanks for the hint, I tried to override the description, however I'm not able to reference the particular step. I tried various options like `jobs.pr.steps[5].with.body` (adds a new property `step[5]`) or `jobs.pr.steps.with.body` (does nothing) but they all either change nothing or produce invalid JSON. When I debug how the path is used, it doesn't look like array access in yaml is supported, but maybe there is a valid notation that I'm missing. 

Since you liked the feature request, being able to configure the commit message would also be helpful. It is not as essential for us as the description though.

**Comment by mrgrain** [comment]
`pullRequestTitle` is already supported. Maybe that should just be used for the commit message?

**Comment by mrgrain** [comment]
IDK, this works for me. Not sure if it's just the different number.

```ts
project.github?.tryFindWorkflow('upgrade')?.file?.patch(JsonPatch.replace('/jobs/pr/steps/4/with/body', 'banana bread'));
```

**Comment by sgrell** [comment]
Thank you very much for your help, using `patch` worked, for some reason only with 5 in the path for me.

**Comment by mrgrain** [comment]
Thinking about it, I believe the overrride syntax is not using `[]`, so this should also work:
```
updateYml?.addOverride('jobs.pr.steps.5..with.body', description);
```

**Comment by sgrell** [comment]
Yes, that notation works as well, I just tried it out. I was certain, that I had tried that, but obviously not.
